src/main/java/com/exadel/bookService/service/BookService.java

```diff
@@ -0,0 +1,88 @@
+package com.exadel.bookService.service;
+
+import com.exadel.bookService.exception.BookNotFoundException;
+import com.exadel.bookService.model.Book;
+import com.exadel.bookService.repository.BookRepository;
+import com.exadel.bookService.validation.BookValidator;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class BookService {
+
+    private final BookRepository repository;
+    private final BookValidator validator;
+
+    public BookService(BookRepository repository, BookValidator validator) {
+        this.repository = repository;
+        this.validator = validator;
+    }
+
+    public List<Book> getAllBooks() {
+        return repository.findAll();
+    }
+
+    public Optional<Book> getBookById(Long id) {
+        return repository.findById(id);
+    }
+
+    public Book createBook(Book book) {
+        validator.validate(book);  // validação isolada
+        return repository.save(book);
+    }
+
+    public Book updateBook(Long id, Book book) {
+
+        System.out.println("Received ISBN: '" + book.getIsbn() + "'");
+
+        Book existing = repository.findById(id)
+                .orElseThrow(() -> new BookNotFoundException("Book not found"));
+
+        validator.validateForUpdate(existing, book);
```

Comment:
I think you should not validate existing book. :) If this book was in you DB it mean it was (should be) validated before persisting. It is better to validate the Book you got as argument